graftm/create.py

```diff
@@ -830,9 +821,23 @@ def main(self, **kwargs):
         temp_output = tempdir.TempDir()
         graftM_graft_test_dir_name = temp_output.name
         temp_output.dissolve()
-        cmd = "graftM graft --forward %s --graftm_package %s --output_directory %s" %(
-            sequences, output_gpkg_path, graftM_graft_test_dir_name)
-        extern.run(cmd)
+        # Take a subset of sequences for testing
+        with tempfile.NamedTemporaryFile(suffix=".fa") as tf:
+            count = 0
+            for name, seq, _ in SequenceIO().each(open(sequences)):
+                count += 1
+                if count < 10:
+                    tf.write(">")
+                    tf.write(s.name)
+                    tf.write("\n")
+                    tf.write(s.seq)
+                    tf.write("\n")
+                else:
+                    break
```

Comment:
Ok thanks, if you're happy you can merge

Joel

On 19 May 2016 at 23:17, Ben J Woodcroft <notifications@github.com> wrote:

> In graftm/create.py
> <https://github.com/geronimp/graftM/pull/208#discussion_r63875711>:
>
> > -        cmd = "graftM graft --forward %s --graftm_package %s --output_directory %s" %(
> > -            sequences, output_gpkg_path, graftM_graft_test_dir_name)
> > -        extern.run(cmd)
> > +        # Take a subset of sequences for testing
> > +        with tempfile.NamedTemporaryFile(suffix=".fa") as tf:
> > +            count = 0
> > +            for name, seq, _ in SequenceIO().each(open(sequences)):
> > +                count += 1
> > +                if count < 10:
> > +                    tf.write(">")
> > +                    tf.write(s.name)
> > +                    tf.write("\n")
> > +                    tf.write(s.seq)
> > +                    tf.write("\n")
> > +                else:
> > +                    break
>
> ok done
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/geronimp/graftM/pull/208/files/885927c5a08012778b42513e7590e4247c4b717a#r63875711>
>